match/lib/match/encrypt.rb

```diff
@@ -3,9 +3,11 @@
 
 module Match
   class Encrypt
+    require 'base64'
+    require 'openssl'
```

Comment:
if we start depending on a new gem, we should modify the Gemfile. If we want to enforce a way to use it, maybe we should provide a wrapper.

Reply:
Which new gem? OpenSSL? It's already used in most places where we want to work with SSL related stuff and we want to use the latest version possible, so we have the latest security updates.
We don't want to enforce the way it's used in this current pull request because that's deprecated, it's just the way how the command line openssl command worked.
After this PR is merged, we can work out a better way of encrypting stuff (generate a random IV instead of generating it from the password, like the key, and provide it unencrypted. Reasons why this is more secure is explained in the links in the PR) and do all new encrypting that way, while still supporting decrypting with the old algorithm if the users' match repo haven't been encrypted in the new way yet.
The main goal for this PR is to solve the error for people who installed OpenSSL 1.1.x on their machine (and work with people who didn't) because that changes the default hashing algorithm and it wasn't implicitly told to the old openssl command to use MD5.

Reply:
@lacostej Any work necessary here or is this good?

Reply:
@Cyberbeni you're right. It's not new. It's just not yet depended upon in the Gemfile. Weird.

Question: are we 100% sure no one managed to encrypt data with openssl 1.1 and its new hashing algorithm and thus wouldn't be able to decrypt it with the previous hash? If so, shouldn't we use a decrypt mechanism that tries with both hash methods?

Reply:
@lacostej it doesn't hurt to fallback to SHA256 if decrypting with MD5 fails.
could only edit in browser, hopefully no errors.

Reply:
@Cyberbeni it looks OK. I would probably add a few notes in the code

* that we use MD5 as primary for encryption even if known to not be the latest/most secure and that this whole encrypt/decrypt algorithm is meant to be improved
* that we fallback to decryption using SHA256 in case people encrypted data using openssl 1.1.0c or later with fastlane <= 2.80.0

or something like that.

Last question: given that the `pkcs5_keyivgen` method is deprecated (https://apidock.com/ruby/v1_9_3_125/OpenSSL/Cipher/pkcs5_keyivgen), was there a non deprecated way to achieve the same results?

Reply:
@lacostej They recommend using OpenSSL::KDF::pbkdf2_hmac instead which has almost the same parameters and only 1 output. pkcs5_keyivgen generates both the key(16 length) and IV(32 length), I tried with same inputs and 16 length and it resulted in a different key, so my guess is that there is no other, non-deprecated method that keeps us backwards compatible with older fastlane versions.

I'll add comments then.